File: Week_1/Day_2/MaxAvgSubArray.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        # Sum of first window
        window_sum = sum(nums[:k])
        max_sum = window_sum

        # Slide the window
        left = 0
        for right in range(k, len(nums)):
            window_sum += nums[right]      # add new element
            window_sum -= nums[left]       # remove old element
            left += 1

            max_sum = max(max_sum, window_sum)

        return max_sum / k

File: Week_1/Day_2/MaxAvgSubArray.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int windowSum = 0;

        // First window sum
        for (int i = 0; i < k; i++) {
            windowSum += nums[i];
        }

        int maxSum = windowSum;
        int left = 0;

        // Slide the window
        for (int right = k; right < nums.length; right++) {
            windowSum += nums[right];   // add new element
            windowSum -= nums[left];    // remove old element
            left++;

            maxSum = Math.max(maxSum, windowSum);
        }

        return (double) maxSum / k;
    }
}